tools/frontend/llvm-cf-handshake.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+set -e
+
+DYNAMATIC_PATH=$1
+
+DYNAMATIC_BINS=$DYNAMATIC_PATH/build/bin
+
+LLVM=$DYNAMATIC_PATH/polygeist/llvm-project 
+
+LLVM_BINS=$LLVM/build/bin
+
+# Example: "dynamatic/integration-test/fir/fir.c"
+F_SRC=$2
+
+[ -f "$F_SRC" ] || exit 1
+
+# Will be change to standard path in the future (i.e., out/comp).
+OUT=/tmp/dhls-frontend-output
+rm -rf $OUT
+mkdir -p $OUT
+
+# NOTE:
+# - ffp-contract will prevent clang from adding "fused add mul" into the IR
+$LLVM_BINS/clang -O0 -S -emit-llvm $F_SRC \
+  -I "$DYNAMATIC_PATH/include"  \
+  -Xclang \
+  -ffp-contract=off \
+  -o $OUT/clang.ll
+
+# NOTE: when calling clang with "-ffp-contract=off", clang will bypass the
+# "-disable-O0-optnone" flag and still adds "optnone" to the IR. This is a
+# hacky way to ignore it
+sed -i "s/optnone//g" $OUT/clang.ll
+
+# NOTE:
+# - instcombine seems optional;
+# - the optnone attribute sliently disables all the optimization in the passes;
+$LLVM_BINS/opt -S \
+  -passes="mem2reg,instcombine,loop-rotate,consthoist,simplifycfg" \
+  -strip-debug \
+  $OUT/clang.ll \
+  > $OUT/clang_optimized.ll
+
+$LLVM_BINS/mlir-translate \
+  --import-llvm $OUT/clang_optimized.ll \
+  > $OUT/clang_optimized_translated.mlir
+
+# Dropping the functions that are not needed in HLS compilation:
+# NOTE: this list is not exhausive.
+$DYNAMATIC_BINS/drop-functions \
+  $OUT/clang_optimized_translated.mlir \
+  --func main \
+  --func rand \
+  --func srand \
```

Comment:
I think only picking up the kernel-named function is straightforward, what do you think?

(Maybe it's difficult to get the kernel name, but in the `compile.sh` we have access to it)

Reply:
Also, could you implement this as a pass? if adding a pass is a bit involved, it might be better to do it in a separate PR.